From 5ef2bfb82fbe29beb4b80fe63c9a90f2e4e0ad5c Mon Sep 17 00:00:00 2001
From: Darren Etheridge <detheridge@ti.com>
Date: Tue, 24 Jan 2017 16:57:11 -0500
Subject: [PATCH 192/205] ARM: dts: am33xx: add DT node for gpu

Add the node into the am33xx.dtsi file for the SGX GPU.

Note: The SGX is disabled by default because it is an optional module
and only some AM335x variants contain this module, such as AM3358 and
AM3357.  The status should be overwritten as "OK" at the corresponding
board.dts

Signed-off-by: Darren Etheridge <detheridge@ti.com>
Signed-off-by: Eric Ruei <e-ruei1@ti.com>
(cherry picked from commit 812961044520a58aacf8a7e0fdf2ba6aec8f9402)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/boot/dts/am335x-evm.dts   |   24 ++++++++++++++++++++++++
 arch/arm/boot/dts/am335x-evmsk.dts |    4 ++++
 arch/arm/boot/dts/am33xx.dtsi      |   16 ++++++++++++++++
 3 files changed, 44 insertions(+)

diff --git a/arch/arm/boot/dts/am335x-evm.dts b/arch/arm/boot/dts/am335x-evm.dts
index ddd8975..a8ef507 100644
--- a/arch/arm/boot/dts/am335x-evm.dts
+++ b/arch/arm/boot/dts/am335x-evm.dts
@@ -793,3 +793,27 @@
 	clocks = <&clk_32768_ck>, <&clkdiv32k_ick>;
 	clock-names = "ext-clk", "int-clk";
 };
+
+&wkup_m3_ipc {
+	ti,scale-data-fw = "am335x-evm-scale-data.bin";
+};
+
+&pruss_soc_bus {
+	status = "okay";
+
+	pruss: pruss@4a300000 {
+		status = "okay";
+
+		pru0: pru@4a334000 {
+			status = "okay";
+		};
+
+		pru1: pru@4a338000 {
+			status = "okay";
+		};
+	};
+};
+
+&sgx {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/am335x-evmsk.dts b/arch/arm/boot/dts/am335x-evmsk.dts
index 9ba4b18..2fd353a 100644
--- a/arch/arm/boot/dts/am335x-evmsk.dts
+++ b/arch/arm/boot/dts/am335x-evmsk.dts
@@ -721,6 +721,10 @@
 	blue-and-red-wiring = "crossed";
 };
 
+&sgx {
+	status = "okay";
+};
+
 &rtc {
 	clocks = <&clk_32768_ck>, <&clkdiv32k_ick>;
 	clock-names = "ext-clk", "int-clk";
diff --git a/arch/arm/boot/dts/am33xx.dtsi b/arch/arm/boot/dts/am33xx.dtsi
index c01fc03..e2eec12 100644
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@ -1125,6 +1125,22 @@
 			reg = <0x48310000 0x2000>;
 			interrupts = <111>;
 		};
+
+		/*
+		 * The SGX is disabled by default because it is an optional
+		 * module and only some AM335x variants contain this module,
+		 * such as AM3358 and AM3357. The status should be overwritten
+		 * as "OK" at the corresponding board.dts.
+		 */
+		sgx: sgx@56000000 {
+			compatible = "ti,am3352-sgx530", "img,sgx530";
+			ti,hwmods = "gfx";
+			reg = <0x56000000 0x10000>;
+			interrupts = <37>;
+			clocks = <&gfx_fck_div_ck>;
+			clock-names = "fclk";
+			status = "disabled";
+		};
 	};
 };
 
-- 
1.7.9.5

